docs/content/v0.20/architecture.md

```diff
@@ -1,36 +1,479 @@
 # KubeStellar Architecture
 
-**NOTE**: This is only a very rudimentary start.
+KubeStellar provides multi-cluster deployment of Kubernetes objects, controlled by a simple 
+placement policy, where Kubernetes objects are expressed in their native format with no wrapping
+or bundling. The high-level architecture for KubeStellar is illustrated in Figure 1.
 
-## Higher Level
+<figure>
+  <img src="assets/high-level-architecture.png"  alt="High Level Architecture">
+  <figcaption align="center">Figure 1 - High Level Architecture </figcaption>
+</figure>
 
-At the higher level there are three modules.
+KubeStellar relies on the concept of *spaces*, which are abstractions 
+that represent a Kubernetes API Server and its controllers. Users can use 
+spaces to perform these tasks:
 
-1. KubeStellar
-2. [the status addon](https://github.ibm.com/dettori/status-addon)
-3. KubeFlex
+1. Create *Workload Definition Spaces* (WDSs) to store the definitions of their workloads.
+2. Create *Inventory and Transport Spaces* (ITSs) to manage the inventory of clusters and 
+the transport of workloads.
+3. Register and label Workload Execution Clusters (WECs) with the Inventory and 
+Transport Space, to keep track of the available clusters and their characteristics.
+4. Define Placement Policies to specify *what" workloads and *where* should be 
+deployed on the WECs.
+5. Submit Workload Definitions in the native Kubernetes format to the WDSs, 
+and let the Placement Policies govern which WECs should receive them.
+6. Check the status of submitted workloads from the WDS.
 
-The last two are independent and KubeStellar depends on them.
+KubeStellar users can have different roles and responsibilities. 
+For example, a System Admin can set up a KubeStellar instance, 
+create WDSs and ITSs, and register WECs. An Application Owner can 
```

Comment:
I think that creating a KubeStellar instance is a much heaver-weight thing that creating a WDS; I would not put them together in the same role.

Since we are not trying to agree here on roles, let's not list hypothetical roles that will just cause arguments. I suggest instead just listing the things that a "user" does.

Reply:
Sounds good, I will change to be less specific